src/testdrive/src/action/kafka/ingest.rs

```diff
@@ -402,13 +408,21 @@ impl Action for IngestAction {
                     false,
                 )?;
                 let mut row = row.as_bytes();
-                let key = match &key_transcoder {
-                    None => None,
-                    Some(kt) => kt.transcode(&mut row)?,
+                let key = if self.omit_key {
+                    None
+                } else {
+                    match &key_transcoder {
+                        None => None,
+                        Some(kt) => kt.transcode(&mut row)?,
+                    }
+                };
```

Comment:
```suggestion
                match (self.omit_key, &key_transcoder) {
                    (true, _) => None,
                    (false, None) => None,
                    (false, Some(kt)) => kt.transcode(&mut row)?
                }
```

something like this should work for the value transcoder too, but that one is simpler so your call